deepchem/models/torch_models/hf_models.py

```diff
@@ -0,0 +1,407 @@
+            df.to_csv(tmpfile.name)
+            loader = dc.data.CSVLoader(["task1"], feature_field="smiles",
+                                     featurizer=dc.feat.DummyFeaturizer())
+            dataset = loader.create_dataset(tmpfile.name)
+
+        # pretraining
+        from deepchem.models.torch_models.hf_models import HuggingFaceModel
+        from transformers.models.roberta import RobertaForMaskedLM, RobertaModel, RobertaConfig
+        config = RobertaConfig(vocab_size=tokenizer.vocab_size)
+        model = RobertaForMaskedLM(config)
+        hf_model = HuggingFaceModel(model=model, tokenizer=tokenizer, task='pretraining', model_dir='model-dir')
+        hf_model.fit(dataset, nb_epoch=1)
+
+        # finetuning a regression model
+        from transformers.models.roberta import RobertaForSequenceClassification
+        config = RobertaConfig(vocab_size=tokenizer.vocab_size, problem_type='regression', num_labels=1)
+        model = RobertaForSequenceClassification(config)
+        hf_model = HuggingFaceModel(model=model, tokenizer=tokenizer, task='finetuning', model_dir='model-dir')
+        hf_model.load_from_pretrained()
+        hf_model.fit(dataset, nb_epoch=1)
+        hf_model.predict(dataset)  # prediction
+        hf_model.evaluate(dataset, metrics=dc.metrics.Metric(dc.metrics.mae_score))  # evaluation
+
+        # finetune a classification model
+        # making dataset suitable for classification
+        import numpy as np
+        y = np.random.choice([0, 1], size=dataset.y.shape)
+        dataset = dc.data.NumpyDataset(X=dataset.X, y=y, w=dataset.w, ids=dataset.ids)
+
+        from transformers import RobertaForSequenceClassification
+        config = RobertaConfig(vocab_size=tokenizer.vocab_size)
+        model = RobertaForSequenceClassification(config)
+        hf_model = HuggingFaceModel(model=model, task='finetuning', tokenizer=tokenizer)
+        hf_model.fit(dataset, nb_epoch=1)
+        hf_model.predict(dataset)
+        hf_model.evaluate(dataset, metrics=dc.metrics.Metric(dc.metrics.f1_score))
+    """
+
+    def __init__(
+            self, model: 'PreTrainedModel', task: str,
+            tokenizer: 'transformers.tokenization_utils.PreTrainedTokenizer',
+            **kwargs):
+        self.task = task
+        self.tokenizer = tokenizer
+        if self.task == 'pretraining':
+            self.data_collator = DataCollatorForLanguageModeling(
+                tokenizer=tokenizer)
+        # Ignoring type. For TorchModel, loss is a required argument but HuggingFace computes
+        # loss during the forward iteration, removing the need for a loss function.
+        super(HuggingFaceModel, self).__init__(
+            model=model,
+            loss=None,  # type: ignore
+            **kwargs)
+
+    def load_from_pretrained(  # type: ignore
+            self, model_dir: Optional[str] = None):
+        """Load HuggingFace model from a pretrained checkpoint
```

Comment:
Can you add a usage example?

Reply:
Can you also add a discussion of how this compares to HF's load from pretrained?

Reply:
added usage example and comparison